src/game/AI/ScriptDevAI/scripts/world/scourge_invasion.cpp

```diff
@@ -945,35 +1055,58 @@ struct ScourgeMinion : public CombatAI
+    std::unordered_map<ObjectGuid, uint32> m_flameshockFollowers;
 
     PallidHorrorAI(Creature* creature) : CombatAI(creature, 999)
     {
+        AddCustomAction(25, 2500u, [&]() // Hacky solution for flameshockers to remain in formation
+        {
+            for (std::pair<ObjectGuid, uint32> follower : m_flameshockFollowers)
+            {
+                if (Unit* f = m_creature->GetMap()->GetUnit(follower.first))
+                {
+                    if (!f->IsInCombat() && f->GetMotionMaster()->GetCurrentMovementGeneratorType() != FOLLOW_MOTION_TYPE)
+                    {
+                        float angle = (float(follower.second) * (M_PI / (m_flameshockFollowers.size() / static_cast<float>(2)))) + m_creature->GetOrientation();
+                        f->GetMotionMaster()->Clear(true, true);
+                        f->GetMotionMaster()->MoveFollow(m_creature, 2.5f, angle);
+                    }
+                }
+            }
+            ResetTimer(25, 2500u);
+        });
         uint32 amount = urand(5, 9); // sniffed are group sizes of 5-9 shockers on spawn.
 
         if (m_creature->GetHealthPercent() == 100.0f)
         {
             for (uint32 i = 0; i < amount; ++i)
             {
-                if (Creature* pFlameshocker = m_creature->SummonCreature(NPC_FLAMESHOCKER, m_creature->GetPositionX(), m_creature->GetPositionY(), m_creature->GetPositionZ(), 0.0f, TEMPSPAWN_TIMED_OR_DEAD_DESPAWN, HOUR * IN_MILLISECONDS, true))
+                if (Creature* pFlameshocker = m_creature->SummonCreature(NPC_FLAMESHOCKER, m_creature->GetPositionX(), m_creature->GetPositionY(), m_creature->GetPositionZ(), 0.0f, TEMPSPAWN_TIMED_OOC_OR_CORPSE_DESPAWN, HOUR * IN_MILLISECONDS, true))
                 {
                     float angle = (float(i) * (M_PI / (amount / static_cast<float>(2)))) + m_creature->GetOrientation();
                     //pFlameshocker->JoinCreatureGroup(m_creature, 5.0f, angle - M_PI, OPTION_FORMATION_MOVE); // Perfect Circle around the Pallid.
+                    pFlameshocker->GetMotionMaster()->Clear(true, true);
+                    pFlameshocker->GetMotionMaster()->MoveFollow(m_creature, 2.5f, angle);
                     pFlameshocker->CastSpell(pFlameshocker, SPELL_MINION_SPAWN_IN, TRIGGERED_OLD_TRIGGERED);
                     m_flameshockers.insert(pFlameshocker->GetObjectGuid());
+                    m_flameshockFollowers.insert(std::pair<ObjectGuid, uint32>(pFlameshocker->GetObjectGuid(), i));
                 }
             }   
         }
         m_creature->SetCorpseDelay(10); // Corpse despawns 10 seconds after a crystal spawns.
         AddCombatAction(EVENT_PALLID_RANDOM_YELL, 5000u);
         AddCombatAction(EVENT_PALLID_SPELL_DAMAGE_VS_GUARDS, 5000u);
         AddCombatAction(EVENT_PALLID_SUMMON_FLAMESHOCKER, 5000u);
+        AddCombatAction(25, 5000u);
+        m_creature->GetMap()->SetWeather(m_creature->GetZoneId(), WEATHER_TYPE_STORM, 0.25f, true);
```

Comment:
I'm not sure this is the right place to put it, but when the capitals get invaded the weather definitely changes to a very dark sky